core/src/dird/authenticate_console.cc

```diff
@@ -354,17 +354,16 @@ static void LogErrorMessage(std::string console_name, UaContext* ua)
         ua->UA_sock->port());
 }
 
-static bool NumberOfConsoleConnectionsExceeded()
+static uint32_t CurrentNumberOfConsoleConnections()
 {
   JobControlRecord* jcr;
-  unsigned int cnt = 0;
+  uint32_t cnt = 0;
 
   foreach_jcr (jcr) {
     if (jcr->is_JobType(JT_CONSOLE)) { cnt++; }
   }
   endeach_jcr(jcr);
-
-  return (cnt >= me->MaxConsoleConnections) ? true : false;
+  return cnt;
 }
```

Comment:
The problem with this function is that it can return wrong results if there is a lot of job contention.
There is a really ugly way to prevent this:

```C++
static std::uint32_t CurrentNumberOfConsoleConnections()
{
  JobControlRecord* jcr;

  std::vector<JobControlRecord*> vec;
  foreach_jcr (jcr) {
    if (jcr->is_JobType(JT_CONSOLE)) {
     // keep the jcr alive
      jcr->IncUseCount();
      vec.push(jcr);
    }
  }
  endeach_jcr(jcr);

  for (auto* jcr : vec) {
   // this looks weird but its just the opposite of IncUseCount()
    FreeJcr(jcr);
  }

  constexpr std::uint32_t max = std::numeric_limits<std::uint32_t>::max();
  if (vec.size() > max) { return max; }
  return vec.size();
}
```
This stops jcrs from getting destroyed after we have counted them.

Reply:
We could even be a bit less pessimistic and say
```
std::size_t ended = 0;
for (auto* jcr : vec) {
  if (jcr->UseCount() == 1) {
     ended += 1;
  }
  FreeJcr(jcr);
}

...

std::size_t num_counted = vec.size();
std::size_t at_most_alive = num_counted - ended;
if (at_most_alive > max) return max;
return at_most_alive;
```

Reply:
Actually this does not really help the situation.